agent/engine/dockerclient/dockerclientfactory.go

```diff
@@ -14,13 +14,21 @@
 package dockerclient
 
 import (
-	"errors"
-
 	"github.com/aws/amazon-ecs-agent/agent/engine/dockeriface"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
 	log "github.com/cihub/seelog"
 	docker "github.com/fsouza/go-dockerclient"
+	"github.com/pkg/errors"
 )
 
+const (
+	// minDockerAPILinux is the min Docker API version supported by agent
+	minDockerAPILinux = Version_1_17
```

Comment:
I don't think this would work on both `linux` and `windows`. 
Please ensure that this change works well on all supported platforms.